### ISSUE TITLE
Cancel subscription at gateway when renewal payments have failed

### ISSUE THREAD

**Issue by thomasfw** [restrictcontentpro/restrict-content-pro#1683]
1. The user's subscription is set to auto-renew, but the renewal payments fail.
2. The user's status is set to `expired` once their expiry date is reached.
3. The user still has a dormant subscription in the gateway (Stripe) with 'Past Due' status, as it was never actively cancelled.

The member's status is set to `expired`, and the 'Cancel' link for this member in Restrict > Members no longer appears, so we have to go to the gateway to manually cancel the unpaid sub.

Can the dormant subscription(s) be cleared in this event?

**Comment by thomasfw** [comment]
Somewhat related. We had an email from a member who couldn't cancel their subscription once the auto-renewal process had started. I think they were unable to cancel from their profile once the first payment had failed due to their `expired` status.

Stripe kept trying to take the payment over the course of the week as expected, but they had no way of cancelling and halting these charge requests, so had to contact us to resolve manually.

**Comment by mindctrl** [comment]
I agree with @nosegraze. We definitely have people that try to recover past due accounts by setting their Stripe account settings to not cancel subs after x # of failed attempts. I like the suggestion to fix the inability to cancel manually.

**Comment by mindctrl** [comment]
`RCP_Member::can_cancel()` checks that the user's status isn't expired, which makes sense for the most part. Trying to cancel a subscription that isn't active is going to result in errors from the gateway APIs, that we'll need to address in different ways for each gateway depending on if they throw exceptions or some other error.

The problem with always showing the cancel link is that for most cases it's not appropriate. When we have the new subscription table from #774 we'll have the cancellation date and will know whether or not the subscription is expired/cancelled by looking at the `cancellation_date` column. We could also introduce a `failing` status and check that instead.

Putting this in the 3.0.1 milestone for now.
